include/etiss/fault/Misc.h

```diff
@@ -0,0 +1,129 @@
+/**
+
+        @copyright
+
+        <pre>
+
+        Copyright 2018 Infineon Technologies AG
+
+        This file is part of ETISS tool, see <https://github.com/tum-ei-eda/etiss>.
+
+        The initial version of this software has been created with the funding support by the German Federal
+        Ministry of Education and Research (BMBF) in the project EffektiV under grant 01IS13022.
+
+        Redistribution and use in source and binary forms, with or without modification, are permitted
+        provided that the following conditions are met:
+
+        1. Redistributions of source code must retain the above copyright notice, this list of conditions and
+        the following disclaimer.
+
+        2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions
+        and the following disclaimer in the documentation and/or other materials provided with the distribution.
+
+        3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse
+        or promote products derived from this software without specific prior written permission.
+
+        THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+        WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+        PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+        DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+        PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+        HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+        NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+        POSSIBILITY OF SUCH DAMAGE.
+
+        </pre>
+
+        @author Chair of Electronic Design Automation, TUM
+
+        @date November 18, 2021
+
+        @version 0.1
+
+*/
+/**
+        @file
+
+        @brief general helpers for fault
+
+        @detail
+
+
+
+
+*/
+
+#ifndef ETISS_INCLUDE_FAULT_MISC_H_
+#define ETISS_INCLUDE_FAULT_MISC_H_
+
+#include <map>
+#include <string>
+#include <vector>
+
+namespace etiss
+{
+namespace fault
+{
+
+////////////////////////////////////////////////////////////////////////////////////////////////
+/// \brief Action type class
+template <typename enum_t>
+class SmartType : public etiss::ToString
```

Comment:
Sure, I understand these concerns. I had this laying around for when I needed string->enum and enum->string encoding as we need here for all the from and to xml stuff going on in ETISS-FI.
The `better-enums` seems perfect, but would need a third party integration of it in ETISS, which I would like to avoid in this PR. I'd even suggest to add a `/third_party/` directory to ETISS root to easier integrate such things - either through git submodules or CMake fetch. Otherwise we end up with missing copyright/licenses and pasted source code like we already have here:
- https://github.com/tum-ei-eda/etiss/tree/master/src/fault/xml
- https://github.com/tum-ei-eda/etiss/tree/master/src/fpu
- https://github.com/tum-ei-eda/etiss/tree/master/src/IntegratedLibrary/SimpleIni
just some that I know of...